sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -72,7 +95,14 @@ class UnivocityParser(
     new CsvParser(parserSetting)
   }
 
-  private val row = new GenericInternalRow(requiredSchema.length)
+  // The row is used as a temporary placeholder of parsed and converted values.
+  // It is needed for applying the pushdown filters.
+  private val parsedRow = new GenericInternalRow(parsedSchema.length)
+  // Pre-allocated Seq to avoid the overhead of the seq builder.
+  private val requiredRow = Seq(new GenericInternalRow(requiredSchema.length))
```

Comment:
So, per https://github.com/apache/spark/pull/26973/files#r366151938, it can be `Option` I believe.